specification/sections/02-providers.md

```diff
@@ -26,6 +26,22 @@ Providers are the "translator" between the flag evaluation calls made in applica
 provider.getMetadata().getName(); // "my-custom-provider"
 ```
 
+#### Requirement 2.1.2
+
+> The provider `metadata` member or accessor **MAY** define an `allFlagMetadata` field or accessor of type collection of [flag_metadata](../types.md#flag-metadata), which identifies all available flag keys in the provider.
+
+```typescript
+provider.getMetadata().getAllFlagMetadata(); // [{"key": "featureA", "type": "boolean"}, {"key": "featureB", "type": "string"}]
```

Comment:
It might be too late to consider, but there was an old OFEP [1] [2] on flag metadata, which is similar to the use cases that attempt to address through this spec change. 

From OFEP pov, `getFlagMetadata(KEY)` & `getAllFlagMetadata()` were new APIs of OF client. 

[1] - https://github.com/open-feature/ofep/pull/44 
[2] - https://github.com/Kavindu-Dodan/ofep/blob/460112b8585e045b2e8fd4bfba83eb7cb22d0990/OFEP-flag-metadata-to-applications.md

Reply:
Oh cool, thank you for linking these, I hadn't seen them. Yeah, I agree with you and @beeme1mr that we probably should move those to be exposed on the client so that users don't need to interact directly with the provider. Let me make that change and see how it looks.